include/boost/static_string/static_string.hpp

```diff
@@ -5614,6 +5614,33 @@ operator+(
     std::size_t(0), +lhs);
 }
 
+//------------------------------------------------------------------------------
+//
+// erase_if
+//
+//------------------------------------------------------------------------------
+
+template<
+    std::size_t N, typename CharT,
+    typename Traits, typename UnaryPredicate>
+BOOST_STATIC_STRING_CPP14_CONSTEXPR
+typename // required for C++11
+basic_static_string<N, CharT, Traits>::size_type
+erase_if(
+    basic_static_string<N, CharT, Traits>& str,
+    UnaryPredicate pred)
+{
+  typename basic_static_string<N, CharT, Traits>::size_type count = 0;
+  auto first = str.begin();
+  for (auto it = first; it != str.end(); ++it)
+    if (!pred(*it))
+      *first++ = std::move(*it);
+    else
+      ++count;
```

Comment:
Instead of maintaining `count` we could simply return `str.end() - first`

Reply:
Good point! Done